docs/querying/functions.md

```diff
@@ -130,6 +130,11 @@ delta(cpu_temp_celsius{host="zeus"}[2h])
 
 `deriv(v range-vector)` calculates the per-second derivative of the time series in a range
 vector `v`, using [simple linear regression](https://en.wikipedia.org/wiki/Simple_linear_regression).
+The range vector must have at least two samples in order to perform the calculation. The linear 
+regression is used to calculate a slope and offset for a trend line, and the slope value is used 
+for the derivative calculation. In order to reduce calculation errors, the timestamp of the first 
+sample in the range vector is set as `x=0` within the linear regression algorithm. When `+Inf` or 
+`-Inf` are found in the range vector, the slope and offset value calculated will be `NaN`.
```

Comment:
I think "the slope value is used for the derivative calculation" is redundant. The slope _is_ essentially the derivative. I think the sentence in lines 131/132 says everything the user needs, namely that simple linear regression is used to calculate the derivative.

I would definitely not explain implementation details like the `x=0` here. This might be replaced by other tricks any day, and it's not helping the user to understand internal optimization to improve the numerical precision.

I would propose to only add the explanation for the special cases here, i.e. no value if there is only one in the range, and `NaN` if any of `+Inf`, `-Inf`, or `NaN` is in the range.